onmetal-controller-manager/controllers/networking/natgateway_controller_test.go

```diff
@@ -0,0 +1,278 @@
+		}
+		Expect(k8sClient.Create(ctx, nic)).To(Succeed())
+
+		By("waiting for the nat gateway routing to be updated")
+		Eventually(func(g Gomega) {
+			Expect(k8sClient.Get(ctx, natGatewayKey, natGatewayRouting)).To(Succeed())
+
+			g.Expect(natGatewayRouting.NetworkRef.Name).To(BeEquivalentTo(network.Name))
+			g.Expect(natGatewayRouting.NetworkRef.UID).To(BeEquivalentTo(network.UID))
+
+			g.Expect(natGatewayRouting.Destinations).To(HaveLen(1))
+
+			g.Expect(natGatewayRouting.Destinations[0].Name).To(BeEquivalentTo(nic.Name))
+			g.Expect(natGatewayRouting.Destinations[0].UID).To(BeEquivalentTo(nic.UID))
+			g.Expect(natGatewayRouting.Destinations[0].IPs).To(HaveLen(1))
+			g.Expect(natGatewayRouting.Destinations[0].IPs[0].IP).To(BeEquivalentTo(natGateway.Status.IPs[0].IP))
+		}).Should(Succeed())
+	})
+
+	It("should reconcile the nattateway and routing destinations, with to little ports", func() {
+		portsPerNetworkInterface := int32(1024)
+		By("creating a network")
+		network := &networkingv1alpha1.Network{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace:    ns.Name,
+				GenerateName: "network-",
+			},
+		}
+		Expect(k8sClient.Create(ctx, network)).To(Succeed())
+
+		By("creating a nat gateway")
+		natGateway := &networkingv1alpha1.NATGateway{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace:    ns.Name,
+				GenerateName: "nat-gateway-",
+			},
+			Spec: networkingv1alpha1.NATGatewaySpec{
+				//ToDo
+				Type: networkingv1alpha1.NATGatewayTypePublic,
+				IPFamilies: []corev1.IPFamily{
+					corev1.IPv4Protocol,
+				},
+				IPs: []networkingv1alpha1.NATGatewayIP{
+					{
+						Name: "ip1",
+					},
+					{
+						Name: "ip2",
+					},
+				},
+				NetworkRef: corev1.LocalObjectReference{
+					Name: network.Name,
+				},
+				NetworkInterfaceSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{"foo": "bar"},
+				},
+				PortsPerNetworkInterface: pointer.Int32(portsPerNetworkInterface),
+			},
+		}
+		Expect(k8sClient.Create(ctx, natGateway)).To(Succeed())
+
+		By("waiting for the nat gateway routing to exist with no destinations")
+		natGatewayKey := client.ObjectKeyFromObject(natGateway)
+		natGatewayRouting := &networkingv1alpha1.NATGatewayRouting{}
+		Eventually(func(g Gomega) {
+			err := k8sClient.Get(ctx, natGatewayKey, natGatewayRouting)
+			Expect(client.IgnoreNotFound(err)).NotTo(HaveOccurred())
+			g.Expect(err).NotTo(HaveOccurred())
+
+			g.Expect(metav1.IsControlledBy(natGatewayRouting, natGateway)).To(BeTrue(), "nat gateway routing is not controlled by nat gateway: %#v", natGatewayRouting.OwnerReferences)
+			g.Expect(natGatewayRouting.Destinations).To(BeEmpty())
+		}).Should(Succeed())
+
+		natGatewayBase := natGateway.DeepCopy()
+		ip := netip.IPv4Unspecified()
+		for _, v := range natGateway.Spec.IPs {
+			ip = ip.Next()
+			natGateway.Status.IPs = append(natGateway.Status.IPs, networkingv1alpha1.NATGatewayIPStatus{
+				Name: v.Name,
+				IP: commonv1alpha1.IP{
+					Addr: ip,
+				},
+			})
+		}
```

Comment:
Rather than having logic in a test, patch constants into the `natGateway.Status` - in this case, since you know your test scenario has two IPs, just patch two constant IPs in.